src/plugins/action_create_user/src/createUser.test.ts

```diff
@@ -0,0 +1,39 @@
+// Test suite for the createUser Action -- just confirms that users are written to database.
+
+const Action = require('./createUser')
+
+const testUser = {
+  first_name: 'Carl',
+  last_name: 'Smith',
+  username: 'ceejay',
+  date_of_birth: '1999-12-23',
+  password_hash: 'XYZ1234',
+  email: 'test@sussol.net',
+}
+
+const invalidUser = {
+  first_name: 'Carl',
+  last_name: 'Smith',
+  user_name: 'ceejay',
+  dateOfBirth: '1999-12-23',
+  password_hash: 'XYZ1234',
+  email: 'test@sussol.net',
+}
+
+test('Test: add User to database', () => {
+  return Action.createUser(testUser).then((result: any) => {
+    expect(result).toEqual({
+      status: 'Success',
+      error_log: '',
+    })
+  })
+})
+
+test('Test: Invalid user -- should fail', () => {
+  return Action.createUser(invalidUser).then((result: any) => {
+    expect(result).toEqual({
+      status: 'Fail',
+      error_log: 'There was a problem creating new user.',
+    })
+  })
+})
```

Comment:
all tests passed, but there is one error output that I'll investigate the reason why:
<img width="945" alt="Screen Shot 2020-10-13 at 2 14 40 PM" src="https://user-images.githubusercontent.com/16461988/95804259-a4543100-0d5e-11eb-8a82-bf91f14c4fe5.png">

Reply:
Just quickly -- this is intentional. The test sends invalid data to the action, so it passes by throwing an error ;)

We can probably make it more elegant by changing what the error returns -- I think the whole error object is probably better than `error.stack`, since it allows the calling method to handle it as it sees fit, and I can also adjust this test to accommodate. Will tweak.

Reply:
Okay, I've updated `postgresConnect` to `throw` errors for each method. That way the calling method can handle the error as it sees fit.

In this case, the `try/catch` block in `createUser` handles the error and updates the database accordingly.

Do you think it's too much to change all of these? If so, I can change them back and just leave the `createUser` one.

Personally I think throwing the error up the call chain is the best approach as it gives the calling method control to handle the error as it sees fit (which might vary in different contexts), (e.g write log to database, console output, UI message, etc.) Plus it makes writing error condition unit tests easier.

Reply:
No, I totally agree with that. The solution before was kind of temporary, and now I see why it is a bad one. We can continue like that and work on a errorHandler that should be called within the calls to the DB layer.